spanner/ddl/ast.go

```diff
@@ -181,16 +248,25 @@ func (ct CreateTable) PrintCreateTable(config Config) string {
 	var col []string
 	var colComment []string
 	var keys []string
-	for i, cn := range ct.ColNames {
-		s, c := ct.ColDefs[cn].PrintColumnDef(config)
-		s = "\n    " + s
-		if i < len(ct.ColNames)-1 {
-			s += ","
-		} else {
-			s += " "
+	if config.TargetDb == "experimental_postgres" {
+		for i, cn := range ct.ColNames {
+			s, c := ct.ColDefs[cn].PrintColumnDef(config)
+			s = "\n    " + s
+			if i < len(ct.ColNames)-1 {
+				s += ","
+			} else {
+				s += ","
+			}
+			col = append(col, s)
+			colComment = append(colComment, c)
+		}
+	} else {
+		for _, cn := range ct.ColNames {
```

Comment:
I'm confused: shouldn't this case be the same as the old code?
e.g. where did the "if i < len(ct.ColNames)-1" go to?

Reply:
I have now cleaned up the handling here. It turned out that `if i < len(ct.ColNames)-1` wasn't needed for either dialect and that both the dialects could produce the same output for column definitions. Now, the only difference is how PRIMARY KEY and INTERLEAVE and printed and those are handled below separately.